flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/windows.scala

```diff
@@ -0,0 +1,340 @@
+
+  private var alias: Option[Expression] = None
+
+  /**
+    * Defines the time mode for streaming tables and specifies a time attribute for
+    * batch tables over which the query is evaluated.
+    *
+    * @param timeField time mode for streaming tables and time attribute for batch tables
+    * @return a tumbling group-window on event-time
+    */
+  def on(timeField: Expression): TumblingEventTimeWindow =
+    new TumblingEventTimeWindow(alias, timeField, size)
+
+  /**
+    * Defines the time mode for streaming tables and specifies a time attribute for
+    * batch tables over which the query is evaluated.
+    *
+    * @param timeField time mode for streaming tables and time attribute for batch tables
+    * @return a tumbling group-window on event-time
+    */
+  def on(timeField: String): TumblingEventTimeWindow =
+    on(ExpressionParser.parseExpression(timeField))
+
+  /**
+    * Assigns an alias for this window that the following `select()` clause can refer to in order
+    * to access window properties such as window start or end time.
+    *
+    * @param alias alias for this window
+    * @return this window
+    */
+  def as(alias: Expression): TumblingWindow = {
+    this.alias = Some(alias)
+    this
+  }
+
+  /**
+    * Assigns an alias for this window that the following `select()` clause can refer to in order
+    * to access window properties such as window start or end time.
+    *
+    * @param alias alias for this window
+    * @return this window
+    */
+  def as(alias: String): TumblingWindow = as(ExpressionParser.parseExpression(alias))
+
+  override private[flink] def toLogicalWindow: LogicalWindow =
+    ProcessingTimeTumblingGroupWindow(alias, size)
+}
+
+/**
+  * Tumbling group-window on event-time.
+  */
+class TumblingEventTimeWindow(
+    alias: Option[Expression],
+    time: Expression,
+    size: Expression)
+  extends EventTimeWindow(time) {
+
+  override private[flink] def toLogicalWindow: LogicalWindow =
+    EventTimeTumblingGroupWindow(name.orElse(alias), time, size)
+}
+
+// ------------------------------------------------------------------------------------------------
+// Sliding group windows
+// ------------------------------------------------------------------------------------------------
+
+/**
+  * Partially specified sliding window.
+  *
+  * @param size size of the window either as number of rows or interval of milliseconds
+  */
+class SlideWithSize(size: Expression) {
+
+  /**
+    * Partially specified sliding window.
+    *
+    * @param size size of the window either as number of rows or interval of milliseconds
+    */
+  def this(size: String) = this(ExpressionParser.parseExpression(size))
+
+  /**
+    * Specifies the window's slide. This is the size of the overlap.
+    *
+    * For example, you could have windows of size 10 minutes that slide by 5 minutes. With this
+    * the following operation gets 10 minutes worth of elements and will be invoked for every 5
```

Comment:
I changed the example to: "For example, you could have windows of size 15 minutes that slide by 3 minutes. With this 15 minutes worth of elements are grouped every 3 minutes."